redis-om-spring/src/main/java/com/redis/om/spring/RedisModulesConfiguration.java

```diff
@@ -171,9 +172,12 @@ public Criteria<Image, DetectedObjects> faceDetectionModelCriteria( //
 
   @Bean(name = "djlFaceDetectionModel")
   public ZooModel<Image, DetectedObjects> faceDetectionModel(
-      @Qualifier("djlFaceDetectionModelCriteria") Criteria<Image, DetectedObjects> criteria)
-      throws MalformedModelException, ModelNotFoundException, IOException {
-    return ModelZoo.loadModel(criteria);
+      @Qualifier("djlFaceDetectionModelCriteria") Criteria<Image, DetectedObjects> criteria) {
+    try {
```

Comment:
Let’s use Optional and then return Optional.empty instead of null in case of error , WDYT?

Reply:
I tried Optional initially but it doesn't work with Spring injection, since it injects a NullBean object when not found but not an Optional.empty() for some reason.

Reply:
Did you try to change the return type of the factory method to Optional ?

Reply:
I did, can you try on your end? I kept on getting a class cast exception on the app context creation.

Reply:
FYI, The way I'm testing this is to run it connected to the networks and then run it without connectivity (turn WIFI off, basically), and they should both pass.

Reply:
I just checked with a few Spring experts, no `Optional` for `@Bean` annotated methods, it seems to not be part of the DI chain to handle those.